dotnet/src/Functions/Functions.Prompty/Extensions/PromptyKernelExtension.cs

```diff
@@ -0,0 +1,127 @@
+        //
+        // e.g
+        // file: chat.prompty
+        // ---
+        // name: Contoso Chat Prompt
+        // description: A retail assistant for Contoso Outdoors products retailer.
+        // authors:
+        //   - XXXX
+        // model:
+        //   api: chat
+        //   configuration:
+        //     type: azure_openai
+        //     azure_deployment: gpt - 35 - turbo
+        //     api_version: 2023 - 07 - 01 - preview
+        //   parameters:
+        //     tools_choice: auto
+        //     tools:
+        //     -type: function
+        //       function:
+        //         name: test
+        //         description: test function
+        //         parameters:
+        //           properties:
+        //             location:
+        //               description: The city and state or city and country, e.g.San Francisco, CA
+        //                 or Tokyo, Japan
+        // ---
+        // ... (rest of the prompty content)
+
+        var splits = text.Split(["---"], StringSplitOptions.RemoveEmptyEntries);
+        var yaml = splits[0];
+        var content = splits[1];
+
+        var deserializer = new DeserializerBuilder().Build();
+        var prompty = deserializer.Deserialize<PromptyYaml>(yaml);
+
+        // step 2
+        // create a prompt template config from the prompty object
+        var promptTemplateConfig = new PromptTemplateConfig
+        {
+            Name = prompty.Name, // TODO: sanitize name
+            Description = prompty.Description,
+            Template = content,
+        };
+
+        PromptExecutionSettings defaultExecutionSetting = prompty.Model?.ModelConfiguration?.ModelType switch
+        {
+            ModelType.azure_openai or ModelType.openai => new OpenAIPromptExecutionSettings()
+            {
+                ResponseFormat = prompty.Model?.Parameters?.ResponseFormat == "json_object" ? ChatCompletionsResponseFormat.JsonObject : null,
+                Temperature = prompty.Model?.Parameters?.Temperature ?? 1.0,
+                TopP = prompty.Model?.Parameters?.TopP ?? 1.0,
+                MaxTokens = prompty.Model?.Parameters?.MaxTokens,
+                Seed = prompty.Model?.Parameters?.Seed,
+                ModelId = prompty.Model?.ModelConfiguration?.AzureDeployment,
+            },
+            _ => throw new NotSupportedException($"Model type '{prompty.Model?.ModelConfiguration?.ModelType}' is not supported."),
+        };
+
+        promptTemplateConfig.AddExecutionSettings(defaultExecutionSetting);
+
+        // step 3. add input variables
+        if (prompty.Inputs != null)
+        {
+            foreach (var input in prompty.Inputs)
+            {
+                if (input.Value is string description)
+                {
+                    var inputVariable = new InputVariable()
+                    {
+                        Name = input.Key,
+                        Description = description,
```

Comment:
We support some additional properties, IsRequest, JsonSchema, AllowUnfaceContent

Reply:
The `AllowUnsafeContent` is track in #6033

Reply:
Other fields are tracking in https://github.com/microsoft/prompty/issues/3